compiler/parser/src/fstring.rs

```diff
@@ -67,14 +67,14 @@ impl FStringParser {
                         Some('a') => ConversionFlag::Ascii,
                         Some('r') => ConversionFlag::Repr,
                         Some(_) => {
-                            return Err(if expression[1..].trim().is_empty() {
+                            return Err(if expression.trim().is_empty() {
```

Comment:
without being sure, is all the `trim`ing required? When we match a space we push it into `trailing_seq` so would `expression` ever have starting/trailing whitespace?

Reply:
I think it's required:

```python
>>> f"{    }"
  File "<stdin>", line 1
SyntaxError: f-string: empty expression not allowed
>>> f"{}"
  File "<stdin>", line 1
SyntaxError: f-string: empty expression not allowed
```

Reply:
Ah right, for some reason I completely missed that match on the space character is guarded by `self.documenting`. Still have a hunch these can be refactored away but that's another issue.